e2eAIOK/deltatuner/deltatuner/deltatuner.py

```diff
@@ -12,9 +12,11 @@
 
 SUPPORTED_ALGO = ["auto", "lora", "ssf"]
 
-def optimize(model, tokenizer, algo: str="auto", adapter_name: str="default", deltatuning_args: DeltaTunerArguments=None) -> DeltaTunerModel:
+def optimize(model, tokenizer, adapter_name: str="default", deltatuning_args: DeltaTunerArguments=None, **kwargs) -> DeltaTunerModel:
     if deltatuning_args is None:
-        deltatuning_args = DeltaTunerArguments()
+        deltatuning_args = DeltaTunerArguments(**kwargs)
+
+    algo = deltatuning_args.algo
```

Comment:
I suggested using the kwargs.pop("key_name", None) method if the "key_name" did not appear in the kwargs

Reply:
kwargs is used to initilize deltatuning_args, suppose will not conflict with other args.